openedx/core/djangoapps/programs/utils.py

```diff
@@ -136,19 +138,31 @@ def progress(self, programs=None, count_only=True):
                 progress, and unstarted courses instead of serialized representations
                 of the courses.
 
+            show_expired_runs_as_remaining (bool): Whether in progress runs whose upgrade
+                deadline has expired should be counted as remaining rather than in progress.
+
         Returns:
             list of dict, each containing information about a user's progress
                 towards completing a program.
         """
+        now = datetime.datetime.now(tzlocal())
         progress = []
         programs = programs or self.engaged_programs
         for program in programs:
-            completed, in_progress, not_started = [], [], []
+            completed, in_progress, not_started, expired = [], [], [], []
 
             for course in program['courses']:
                 if self._is_course_complete(course):
                     completed.append(course)
                 elif self._is_course_in_progress(course):
+                    if show_expired_runs_as_remaining:
+                        enrolled_run = [run for run in course['course_runs'] if run['is_enrolled']][0]
```

Comment:
This is only going to work for an "extended_program." In other words, this will only work if you provide extended programs when initializing the class. Expecting people to know that when using this method worries me.

Furthermore, how are these course runs ordered? My sense is that it's not safe to just pick the first one. Consider the case where the user is enrolled in two runs of the same course. This may pick the older enrollment and miss the newer one.